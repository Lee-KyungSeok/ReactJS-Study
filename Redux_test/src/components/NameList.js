import React, { Component } from 'react';

class NameList extends Component {
    static defaultProps = {
        names: []
    };

    renderList() {
        const { names } = this.props;
        return names.map(
            (name, i) => (<li key={i}>{name}</li>)
        );
    }

    render() {
        return (
            <ul>
                { this.renderList() }
            </ul>
        );
    }
}

export default NameList;