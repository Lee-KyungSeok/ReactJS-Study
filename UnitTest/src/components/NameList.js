import React from 'react';

const NameList = ({names}) => {
    const nameList = names.map(
        (name, i) => (
            <li key={i}>{name}</li>
        )
    );

    return(
        <ul>
            {nameList}
        </ul>
    )
};

export default NameList;