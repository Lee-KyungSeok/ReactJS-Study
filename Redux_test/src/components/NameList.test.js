import React from 'react';
import {shallow} from 'enzyme';
import NameList from './NameList';

describe('NameList', () => {
    let component = null;

    it('renders correctly', () => {
        component = shallow(<NameList names={['Kyung','Seok']}/>);
    });

    it('match correctly', () => {
        expect(component).toMatchSnapshot();
    });
});