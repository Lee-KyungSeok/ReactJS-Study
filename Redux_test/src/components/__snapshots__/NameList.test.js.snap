// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameList match correctly 1`] = `
<ul>
  <li
    key="0"
  >
    Kyung
  </li>
  <li
    key="1"
  >
    Seok
  </li>
</ul>
`;
